addons/mail/static/src/new/public/discuss_public_welcome_page.js

```diff
@@ -0,0 +1,76 @@
+/* @odoo-module */
+
+import { Component, useState } from "@odoo/owl";
+import { useMessaging } from "../messaging_hook";
+
+const getNextGuestId = (function () {
+    let id = 0;
+    return () => ++id;
+})();
+
+export class DiscussPublicWelcomePage extends Component {
+    static props = ["channel", "shouldAddGuestAsMemberOnJoin?"];
+    static template = "mail.discuss_public_welcome_page";
+
+    setup() {
+        this.messaging = useMessaging();
+        this.state = useState({
+            pendingGuestName: "",
+        });
+    }
+
+    onKeydownGuestNameInput(ev) {
+        if (ev.key === "Enter") {
+            this.joinChannel();
+        }
+    }
+
+    onClickJoinButton(ev) {
+        this.joinChannel();
+    }
+
+    /**
+     * Updates guest if needed then displays the thread view instead of the
+     * welcome view.
+     */
+    async joinChannel() {
+        if (this.messaging.state.currentGuest) {
+            await this.messaging.notifyGuestNameToServer({
+                guest_id: this.messaging.state.currentGuest.id,
+                name: this.state.pendingGuestName.trim(),
+            });
+        }
+        if (this.props.shouldAddGuestAsMemberOnJoin) {
+            await this.addCurrentGuestAsMember();
+        }
+        //this.switchToThreadView();
+    }
+
+    /**
+     * Adds the current guest to members of the channel linked to this
+     * welcome view.
+     */
+    async addCurrentGuestAsMember() {
+        await this.messaging.rpc("/mail/channel/add_guest_as_member", {
+            channel_id: this.props.channel.id,
+            channel_uuid: this.props.channel.uuid,
+        });
+    }
+
+    //TODO: should be a component and return as a prop
+    get callDemo() {
+        return this.props.channel && this.props.channel.defaultDisplayMode === "video_full_screen"
+            ? true
+            : false;
+    }
+
+    get guestNameInputUniqueId() {
+        return `o_WelcomePage_guestNameInput_${getNextGuestId()}`;
+    }
```

Comment:
Prefer dumping in DOM rather than indirection in JS.
Also use current convention of kebab-case `o-mail-welcome-page-input`.
The id generator is overkill and useless, so just drop it (makes code simpler)